Add bank balance API

kalkins opened the pull request:
Also has i bugfix and adds a generic view.

Fixes #9

kalkins committed:
Minor fix to url passing in tests

kalkins committed:
Add ByDateRangeView

kalkins committed:
Add bank balance API

Fixes #9

pep8speaks commented:
Hello @kalkins! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`balance/serializers.py`](https://github.com/Kodeworks/liquidator-backend/blob/fc00caac03bd43b6993b058f37c26fbbab6c1235/balance/serializers.py):

> [Line 5:1](https://github.com/Kodeworks/liquidator-backend/blob/fc00caac03bd43b6993b058f37c26fbbab6c1235/balance/serializers.py#L5): [F401](https://duckduckgo.com/?q=pep8%20F401) 'company.models.Company' imported but unused







* In the file [`base/mixins.py`](https://github.com/Kodeworks/liquidator-backend/blob/fc00caac03bd43b6993b058f37c26fbbab6c1235/base/mixins.py):

> [Line 3:1](https://github.com/Kodeworks/liquidator-backend/blob/fc00caac03bd43b6993b058f37c26fbbab6c1235/base/mixins.py#L3): [F401](https://duckduckgo.com/?q=pep8%20F401) 'rest_framework.exceptions.PermissionDenied' imported but unused















##### Comment last updated at 2019-07-01 12:27:37 UTC

kalkins committed:
Add a base serializer and validator for handling foreign keys suffixes

We mostly use a '_id' suffix for foreign keys in the API. Django Rest
Framework doesn't support this by default, at least not with
validators, so I wrote a serializer and validator that handles it.

The serializer should be inherited from by all serializers, except
for User and Company.

kalkins committed:
Add mixins for error handling and limiting querysets to the current company

kalkins committed:
Update bank balance serializer and tests
